src/V3LinkLValue.cpp

```diff
@@ -96,6 +98,24 @@ class LinkLValueVisitor final : public VNVisitor {
             m_setStrengthSpecified = false;
             iterateAndNextNull(nodep->rhsp());
         }
+        if (m_inInitialStatic && m_inFunc) {
+            const bool rhsHasIO = nodep->rhsp() && nodep->rhsp()->exists([](AstNodeVarRef* refp) {
+                // Exclude module I/O referenced from a function/task.
+                return refp->varp() && refp->varp()->isIO()
+                       && refp->varp()->lifetime() != VLifetime::NONE;
+            });
```

Comment:
Thanks for pointing it out. Since Verilator makes input variables `automatic` (as mentioned in 1.), throwing error that this static variable initializer dependence is illegal will not be descriptive enough (end-user cannot do anything about it since inputs are implicitly automatic). 

Therefore, I propose introduction of an unsupported warning for initializing with function/task I/O variables AND, in the remaining cases, error when automatic variable is used as static variable initializer.

@gezalore Commit with proposed approach: https://github.com/verilator/verilator/pull/5296/commits/7ea4eb3ad8339b4476b55d2a08fc2bda1f4280a4